pkg/reconciler/instances/istio/webhooks/webhooks.go

```diff
@@ -0,0 +1,82 @@
+package webhooks
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	retry "github.com/avast/retry-go"
+	"github.com/kyma-incubator/reconciler/pkg/reconciler/instances/istio/clientset"
+	"go.uber.org/zap"
+	"istio.io/istio/istioctl/pkg/tag"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	retriesCount        = 5
+	delayBetweenRetries = 5 * time.Second
+)
+
+var deactivatedLabel = map[string]string{
+	"istio.io/deactivated": "never-match",
+}
+
+// DeleteConflictedDefaultTag deletes conflicted tagged MutatingWebhookConfiguration, if it exists and if the default revision MutatingWebhookConfiguration is not deactivated by istioctl.
+func DeleteConflictedDefaultTag(ctx context.Context, provider clientset.Provider, kubeConfig string, logger *zap.SugaredLogger) error {
+	kubeClient, err := provider.RetrieveFrom(kubeConfig, logger)
+	if err != nil {
+		logger.Error("Could not retrieve KubeClient from Kubeconfig!")
+		return err
+	}
+
+	retryOpts := []retry.Option{
+		retry.Delay(delayBetweenRetries),
+		retry.Attempts(uint(retriesCount)),
+		retry.DelayType(retry.FixedDelay),
+	}
+
+	err = retry.Do(func() error {
+		webhooks, err := tag.GetWebhooksWithTag(ctx, kubeClient, tag.DefaultRevisionName)
+		if err != nil {
+			return err
+		}
+
+		if !isDefaultRevisionDeactivated(ctx, kubeClient) && len(webhooks) > 0 {
+			for _, wh := range webhooks {
```

Comment:
Maybe we can add a comment like this to make it more clear why we can delete this webhooks:
"As the default revision is not deactivated and handles the injection, we are safe to delete all other webhook configurations that were created during the failed installation."

Reply:
I will add the comment